pkg/upgrade/types.go

```diff
@@ -6,24 +6,26 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/openshift/openshift-azure/pkg/api"
-	"github.com/openshift/openshift-azure/pkg/initialize"
 	"github.com/openshift/openshift-azure/pkg/log"
 )
 
 type Upgrader interface {
-	initialize.Initializer
-	Update(ctx context.Context, cs *api.OpenShiftManagedCluster, azuredeploy []byte, config api.PluginConfig) error
+	InitializeCluster(ctx context.Context, cs *api.OpenShiftManagedCluster) error
```

Comment:
at first glance this looks pretty dodgy: healthcheck, initializecluster, waitforinfraservices can all be package-internal now. 
 Deploy and Update can condense to CreateOrUpdate.  The *plugin* external API gets simplified.  I don't think the public interface should expand like this.